apps/backend/src/app/modules/maps/map-review.service.ts

```diff
@@ -1,4 +1,67 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { DbService } from '../database/db.service';
+import { MapReviewDto, PagedResponseDto } from '@momentum/backend/dto';
+import { Role } from '@momentum/constants';
+import { Prisma } from '@prisma/client';
+
+// Using it due to the type inference when including relations
+type ReviewsIncludingReviewers = Prisma.MapReviewGetPayload<{
+  include: { reviewer: true };
+}>;
 
 @Injectable()
-export class MapReviewService {}
+export class MapReviewService {
+  constructor(private readonly db: DbService) {}
+
+  async getReviews(
+    mapID: number,
+    expand: string[],
+    official: boolean
+  ): Promise<PagedResponseDto<MapReviewDto>> {
+    // check if map exists
+    if (!(await this.db.mMap.exists({ where: { id: mapID } })))
+      throw new NotFoundException('Map not found');
+
+    // always including reviewer to check for official/unofficial reviews
+    const expandFiltered = expand?.filter((x) =>
+      ['mmap', 'reviewer'].includes(x)
+    );
+    const include = {};
+
+    if (expandFiltered) for (const x of expandFiltered) include[x] = true;
+
+    if (official !== undefined) include['reviewer'] = true;
+
+    const dbResponse = (await this.db.mapReview.findMany({
+      where: { mapID },
+      include: Object.keys(include).length > 0 ? include : undefined
+    })) as ReviewsIncludingReviewers[];
+
+    // filter by official/unofficial
+    if (official !== undefined) {
+      let filteredResponse;
+      if (official)
+        filteredResponse = dbResponse.filter((x) => {
+          return [Role.ADMIN, Role.MODERATOR, Role.REVIEWER].includes(
+            x.reviewer.roles
+          );
+        });
+      else
+        filteredResponse = dbResponse.filter(
+          (x) =>
+            ![Role.ADMIN, Role.MODERATOR, Role.REVIEWER].includes(
+              x.reviewer.roles
+            )
+        );
+      if (!expand?.includes('reviewer'))
```

Comment:
Worth commenting why we're doing this since it's a bit strange e.g. "We need this `include` for the above logic, but it may not have been on the original `expand` - delete if not."